File: rb/features.rb
# Peremen SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module PeremenFeatures
  def self.make_feature(name)
    case name
    when "base"
      PeremenBaseFeature.new
    when "ratelimit"
      PeremenRatelimitFeature.new
    when "retry"
      PeremenRetryFeature.new
    when "test"
      PeremenTestFeature.new
    when "timeout"
      PeremenTimeoutFeature.new
    else
      PeremenBaseFeature.new
    end
  end
end

File: py/features.py
# Peremen SDK feature factory

from feature.base_feature import PeremenBaseFeature
from feature.test_feature import PeremenTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PeremenBaseFeature(),
        "test": lambda: PeremenTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
